File: app/src/main/java/nz/scuttlebutt/tremola/ssb/peering/discovery/LookUpClient.java
package nz.scuttlebutt.tremola.ssb.peering.discovery;

import android.content.Context;
import nz.scuttlebutt.tremola.ssb.core.SSBid;

public abstract class LookUpClient {

    protected final int port;
    protected final nz.scuttlebutt.tremola.ssb.core.SSBid ed25519KeyPair;
    protected final Context context;
    protected final LookUp lookUp;

    public LookUpClient(LookUp lookUp, Context context, int port, SSBid ed25519KeyPair) {
        this.port = port;
        this.ed25519KeyPair = ed25519KeyPair;
        this.context = context;
        this.lookUp = lookUp;

    }

    abstract void sendQuery(String broadcastMessage);

    // TODO Do I need this?
//    abstract void processQuery();
}

File: app/src/main/java/nz/scuttlebutt/tremola/ssb/peering/discovery/LookupClient.kt
package nz.scuttlebutt.tremola.ssb.peering.discovery

import android.content.Context
import nz.scuttlebutt.tremola.ssb.core.SSBid
import java.util.concurrent.locks.ReentrantLock

abstract class LookupClient(
    protected val lookup: Lookup,
    protected val context: Context,
    protected val ed25519KeyPair: SSBid,
    protected val lock: ReentrantLock
) :
    Thread() {
    var active: Boolean = false
        protected set

    abstract fun sendQuery(broadcastMessage: String)

    /**
     * Close a client when the communication mean is not available.
     */
    fun close() {
        active = false
    }

    /**
     * Allow a client to be used when the communication mean becomes available.
     */
    fun reactivate() {
        active = true
    }

    fun closeQuery(message: String) {}
    val subClass: String
        get() = this.javaClass.toString()
}